### PR TITLE
update(ci): porting CI from test-infra to GHA.

### PR DESCRIPTION
**Pull request by cappellinsamuele**
**What type of PR is this?**

/kind design

**Any specific area of the project related to this PR?**

/area tests

**What this PR does / why we need it**:

This PR is needed to run the CI directly in this repository instead of running it from `test-infra`.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
I'd also remove the build.sh script that is useless after this one :)

**Comment by FedeDP** [comment]
I think we should keep build.sh script until we cleanup the prow job ;)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  check-registry:
```
[the remark this answers is not in the document]
Comment (FedeDP) [reply .github/workflows/ci.yml:8]: UH there is no required status check indeed :D so, i'd use the most meaningful name possible, like 
`run-tests`

[the remark this answers is not in the document]
Comment (cappellinsamuele) [reply .github/workflows/ci.yml:8]: Sure!

[the remark this answers is not in the document]
Comment (FedeDP) [reply .github/workflows/ci.yml]: TODO: we will add the new gha job as required status check on test-infra then ;)

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**